src/main/java/seedu/address/model/person/Year.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's name in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidYear(String)}
+ */
+public class Year {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Year must be a single digit number, that is between 1 and 4!";
```

Comment:
I think we should allow up to Year 6 instead of only 4? Double Degree programmes in NUS goes up to year 5 by default and NUS's maximum candidature is 6 years

```suggestion
            "Year must be a single digit number, that is between 1 and 6!";
```